projects/plugins/jetpack/extensions/blocks/sharing-button/class-sharing-source-block.php

```diff
@@ -503,6 +503,62 @@ public function process_request( $post, array $post_data ) {
 	}
 }
 
+/**
+ * Handle the display of the sms sharing button.
+ */
+class Share_SMS_Block extends Sharing_Source_Block {
+	/**
+	 * Service short name.
+	 *
+	 * @var string
+	 */
+	public $shortname = 'sms';
+
+	/**
+	 * Service name.
+	 *
+	 * @return string
+	 */
+	public function get_name() {
+		return _x( 'SMS', 'as sharing source', 'jetpack' );
+	}
+
+	/**
+	 * Get the URL for the link.
+	 *
+	 * @param int         $post_id         Post ID.
+	 * @param string      $query           Additional query arguments to add to the link. They should be in 'foo=bar&baz=1' format.
+	 * @param bool|string $id              Sharing ID to include in the data-shared attribute.
+	 * @param array       $data_attributes The keys are used as additional attribute names with 'data-' prefix.
+	 *                                     The values are used as the attribute values.
+	 * @return object Link related data (url and data_attributes);
+	 */
+	public function get_link( $post_id, $query = '', $id = false, $data_attributes = array() ) {
+		// We don't need to open new window, so we set it to false.
+		$id = false;
+
+		$post_title = $this->get_share_title( $post_id );
+		$post_url   = $this->get_share_url( $post_id );
+
+		/** This filter is documented in plugins/jetpack/modules/sharedaddy/sharedaddy.php */
+		$sms_subject = sprintf( '[%s] %s', __( 'Shared Post', 'jetpack' ), $post_title );
+
+		$sms_query = sprintf(
+			'?subject=%s&body=%s&share=sms',
+			rawurlencode( $sms_subject ),
+			rawurlencode( $post_url )
```

Comment:
This doesn't quite work for me; only the post URL is shared in the created text message.